instrumentation/testing/testing.go

```diff
@@ -59,7 +77,17 @@ func StartTest(t *testing.T, opts ...Option) *Test {
 
 // Starts a new test with and uses the caller pc info for Name and Suite
 func StartTestFromCaller(t *testing.T, pc uintptr, opts ...Option) *Test {
-	test := &Test{t: t}
+	mutex.Lock()
+	if testPtr, ok := testMap[t]; ok {
+		testPtr.hasEnded = true
+		testPtr.stopCapturingLogs()
+		delete(testMap, testPtr.t)
+	}
```

Comment:
As discussed, let's try to remove this by:
* Deprecating `test.End()` and always calling our own internal implementation of `end()` using autoinstrumentation
* Making the log capture logic idempotent
* Removing the redundant `delete` statement

Reply:
Done without deprecating test.End() only changing the logic to work only in sub tests